### ISSUE TITLE
WinUI 3 Window.ExtendsContentIntoTitleBar caption buttons can be clicked when mouse is not over them

### ISSUE THREAD

**Issue by BreeceW** [microsoft/microsoft-ui-xaml#7259]
### Describe the bug

When a user presses the left mouse button down over a caption button (i.e., Minimize, Restore/Maximize, Close) in the title bar when the title bar has been customized using `Window.ExtendsContentIntoTitleBar`, the button will still be clicked even if the user has moved their mouse off of the button.

### Steps to reproduce the bug

1. Customize a WinUI 3 window where `Window.ExtendsContentIntoTitleBar = true` (you can use the WinUI 3 Gallery app as an example)
2. Hover over the close button (this issue affects minimize and maximize as well), and click the left mouse button down
3. Move the mouse away from the close button (the close button will still appear in the pressed visual state)
4. Release the left mouse button
5. The window will close

### Expected behavior

The button should not be clicked if the left mouse button is released when cursor has moved off the caption button. Additionally, the button should not appear in the pressed visual state if the cursor is not over it.

### Screenshots

The expected behavior when using the standard title bar:

https://user-images.githubusercontent.com/18747724/174465116-860ff8f9-c0cf-4b05-b3f1-ed36243f25b8.mp4

The problematic behavior when using `Window.ExtendsContentIntoTitleBar`:

https://user-images.githubusercontent.com/18747724/174465117-b02bdeac-caec-405f-a568-99c30534f76f.mp4

### NuGet package version

WinUI 3 - Windows App SDK 1.1.1

### Windows app type

- ☐ UWP
- 🗹 Win32

### Device form factor

Desktop

### Windows version

Windows 11 (22H2): Build 22621

### Additional context

This is an important usability issue because these caption buttons behave differently from every other Windows caption buttons and has the potential to cause loss of work. If, for instance, a user inadvertently clicks their mouse button down on the close button, the window will close once they release the mouse button, and there is nothing they can do about it. The correct behavior would allow the user to move the cursor off the close button and safely release the mouse button.

Note that this issue only affects `Window.ExtendsContentIntoTitleBar` and not `AppWindowTitleBar.ExtendsContentIntoTitleBar` or the default title bar.

**Comment by krschau** [comment]
I can also repro this with the WinUI 3 Controls Gallery.
FYI @pratikone

**Comment by everdrone** [comment]
Experiencing this too, is there a fix yet?

**Comment by pratikone** [comment]
Thanks for reporting this bug. I am looking into it.

**Comment by Gabboxl** [comment]
I'm facing the same problem, any update on this?

Also, the captions controls used when setting `Window.ExtendsContentIntoTitleBar = true` don't seem to be consistent compared to native windowing ones.

Instead, the caption controls used when setting `AppWindowTitleBar.ExtendsContentIntoTitleBar = true` with a transparent background with the following code feel very native, just like a UWP app.

```csharp
AppWindowTitleBar titleBar = AppWindow.TitleBar;

titleBar.ExtendsContentIntoTitleBar = true;
titleBar.ButtonBackgroundColor = Colors.Transparent;
titleBar.ButtonInactiveBackgroundColor = Colors.Transparent;
```

**Comment by pratikone** [comment]
I am working to merge WinUI 3 custom titlebar with Appwindow titlebar in a way that winui 3 custom titlebar apis will internally call appwindow apis. so the behavior will start matching once that happens.

**Comment by pratikone** [comment]
It has been fixed in Winappsdk 1.4
